stdlib/Sockets/src/addrinfo.jl

```diff
@@ -234,9 +234,20 @@ ip"fe80::9731:35af:e1c5:6e49"
 """
 function getipaddr(addr_type::Type{T}) where T<:IPAddr
     addrs = getipaddrs(addr_type)
+
     if length(addrs) == 0
         error("No networking interface available")
     end
+
+    # Prefer IPv4 addresses over IPv6 maintaining the ordering provided by libuv
+    sort!(addrs, lt=(x,y) -> begin
+        if x isa IPv4 && y isa IPv6
+            return true
+        else
+            return false
+        end
+    end)
```

Comment:
This should use `findfirst` instead.

Reply:
Were you envisioning something like:
```julia
addr = findfirst(ip -> ip isa IPv4, addrs)
return addr !== nothing ? addr : addrs[1]
```
If there are no IPv4 addresses we want to fall back to the first IPv6 address (or whatever IPAddr is left)

Reply:
Yes.